stumpy/stamp.py

```diff
@@ -126,15 +130,30 @@ def stamp(T_A, T_B, m, ignore_trivial=False):
     T_B.shape[0]-m+1
     """
 
+    # Preprocessing to remove nan and inf values
+    if T_A.ndim != 1:  # pragma: no cover
+        raise ValueError(f"T_A is {T_A.ndim}-dimensional and must be 1-dimensional. ")
+
+    T_A = T_A.copy()
+    T_A[np.isinf(T_A)] = np.nan
     core.check_dtype(T_A)
-    core.check_nan(T_A)
+
+    T_B = T_B.copy()
+    T_B[np.isinf(T_B)] = np.nan
+    if T_B.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T_B is {T_B.ndim}-dimensional and must be 1-dimensional. "
+            "For multidimensional STUMP use `stumpy.mstump` or `stumpy.mstumped`"
+        )
     core.check_dtype(T_B)
-    core.check_nan(T_B)
+
     core.check_window_size(m)
     subseq_T_B = core.rolling_window(T_B, m)
     excl_zone = int(np.ceil(m / 2))
     M_T, Σ_T = core.compute_mean_std(T_A, m)
 
+    T_A[np.isnan(T_A)] = 0
```

Comment:
No there should be no line like this. If a sub-sequence in T_B is nan (or inf, since inf gets replaced with nan in line 142), then the distance profile of this whole sub-sequence has to be set to np.inf. This happens in core.mass, line 485, and that why it is important to keep this information.

Reply:
Maybe I'm missing something but wasn't the point of setting `np.nan` to zero (after the mean and std are computed) so that the dot product wouldn't be NaN? What makes T_A so special/different from T_B that only T_A `np.nan` values are set to zero (after computing the mean and std) but not T_B?

Reply:
Because of the internal workings of `mass`.
It takes a sub-sequence (`Q`, here a subsequence of `T_B`) and calculates the distance profile to whole `T` (here `T_A`). If `Q` is none somewhere, no distance profile is calculated (if statement in core line 485). If T would be none anywhere, then the whole dot product `QT` fails to be computed, so we have to replace any nan values in T. If we would however replace a `nan` in T_B, and would call `mass` with a sub-sequence that should be ignored, we would have no way of telling this to mass, because no other parameter (like mu_Q) is passed to the function.